test/unit/http/auto_retry_test.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2015 Workiva Inc.
```

Comment:
It would also be fine to move the `calculate()` logic into a utility method that could be tested in isolation - in fact that's probably a good idea. Then you could just use that util method in the common request implementation.